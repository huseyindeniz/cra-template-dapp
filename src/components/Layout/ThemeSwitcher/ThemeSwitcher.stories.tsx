// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { ThemeSwitcher } from ".";

export default {
  title: "Layout/ThemeSwitcher",
  component: ThemeSwitcher,
  args: {},
} as ComponentMeta<typeof ThemeSwitcher>;

export const Default: ComponentStory<typeof ThemeSwitcher> = (args) => (
  <ThemeSwitcher {...args} />
);
