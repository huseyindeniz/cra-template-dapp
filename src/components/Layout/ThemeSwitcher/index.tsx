import { Button, useColorMode } from "@chakra-ui/react";
import { IoMdMoon } from "@react-icons/all-files/io/IoMdMoon";
import { IoMdSunny } from "@react-icons/all-files/io/IoMdSunny";

export const ThemeSwitcher: React.FC = () => {
  const { colorMode, toggleColorMode } = useColorMode();

  return (
    <Button onClick={toggleColorMode} variant={"outline"}>
      {colorMode === "light" ? <IoMdMoon /> : <IoMdSunny />}
    </Button>
  );
};
