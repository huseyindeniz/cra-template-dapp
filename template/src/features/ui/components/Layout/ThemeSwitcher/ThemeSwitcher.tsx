import { IconButton, useColorMode } from '@chakra-ui/react';
import { IoMdMoon } from '@react-icons/all-files/io/IoMdMoon';
import { IoMdSunny } from '@react-icons/all-files/io/IoMdSunny';
import { useTranslation } from 'react-i18next';

export const ThemeSwitcher: React.FC = () => {
  const { t } = useTranslation('Layout');
  const { colorMode, toggleColorMode } = useColorMode();

  return (
    <IconButton
      onClick={toggleColorMode}
      role="button"
      variant="outline"
      icon={colorMode === 'light' ? <IoMdMoon /> : <IoMdSunny />}
      aria-label={t('Toggle Color Mode')}
    />
  );
};
