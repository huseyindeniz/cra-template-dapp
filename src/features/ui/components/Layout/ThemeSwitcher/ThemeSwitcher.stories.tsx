// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { ThemeSwitcher } from './ThemeSwitcher';

export default {
  title: 'ui/Components/Layout/ThemeSwitcher',
  component: ThemeSwitcher,
  args: {},
} as ComponentMeta<typeof ThemeSwitcher>;

export const Default: ComponentStory<typeof ThemeSwitcher> = args => (
  <ThemeSwitcher {...args} />
);
