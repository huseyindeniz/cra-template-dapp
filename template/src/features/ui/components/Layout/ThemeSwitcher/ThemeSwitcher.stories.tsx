// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { ThemeSwitcher } from './ThemeSwitcher';

export default {
  title: 'ui/Components/Layout/ThemeSwitcher',
  component: ThemeSwitcher,
  args: {},
} as ComponentMeta<typeof ThemeSwitcher>;

const Template: ComponentStory<typeof ThemeSwitcher> = args => (
  <ThemeSwitcher {...args} />
);

export const Default = Template.bind({});
