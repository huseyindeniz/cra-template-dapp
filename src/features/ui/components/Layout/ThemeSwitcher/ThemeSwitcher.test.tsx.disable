import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import React from 'react';

import * as stories from './ThemeSwitcher.stories';

describe('Feature: UI', () => {
  describe('Component: Layout/ThemeSwitcher', () => {
    const { Default } = composeStories(stories);
    describe('Scenario: Default', () => {
      it('renders a ThemeSwitch button', () => {
        // Arrange
        const { asFragment, getByRole } = render(<Default {...Default.args} />);
        // Assert
        expect(
          getByRole('button', { name: 'Toggle Color Mode' })
        ).toBeInTheDocument();
        expect(asFragment).toMatchSnapshot();
      });
    });
  });
});
