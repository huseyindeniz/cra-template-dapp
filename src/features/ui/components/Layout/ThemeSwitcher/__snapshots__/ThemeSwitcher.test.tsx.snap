// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: UI Component: Layout/ThemeSwitcher Scenario: Default renders a ThemeSwitch button 1`] = `[Function]`;
